src/Microsoft.SqlTools.ServiceLayer/Connection/ConnectionService.cs

```diff
@@ -324,13 +324,15 @@ public ConnectionCompleteParams ValidateConnectParams(ConnectParams connectionPa
             {
                 return new ConnectionCompleteParams
                 {
+                    ErrorMessage = SR.ConnectionServiceConnectErrorNullParams,

```

Comment:
Well this is kind of a mess...

So according to https://github.com/Microsoft/sqlops-dataprotocolclient/blob/c03fa98d19be3c1fba195f861c842f309056b1c0/src/types.ts#L161 (and other matching types like https://github.com/Microsoft/azuredatastudio/blob/466b80fb21351c10632b6e8f5c121cfbe46e1289/src/sql/azdata.d.ts#L415) this is the definition for these two : 

`messages` : `any diagnostic messages return from the service host`
`errorMessage` : `Error message returned from the engine, if any`

Which isn't exactly clear about how these should each be used. Should non-engine errors only go in messages? What are "diagnostic" messages?

Looking in STS it seems like generally ErrorMessage is used to contain just the Message portion of an exception, while ErrorMessage usually contains the full stack trace of the exception. This isn't consistent though...

So as far as this fix goes - what you have is technically correct but we should take the time to figure out exactly what is the correct way to handle these fields and ensure that all our code is using that correctly. And there's two things that'd be good to look into while we're fixing this : 

1. Updating the documentation for the sqlopsclient and related types - they don't seem correct based on how I see the fields from the messages being used
2. Remove the messages field from this error - that shouldn't be necessary to duplicate (and seems like a code smell to me). If it is let's figure out why and whether there's any way we can remove that dependency without breaking anything

Reply:
@alanrenmsft If you could take a quick look over the code and see if I'm missing anything too that'd be great

Reply:
I agree about the clean up comments overall.

@smartguest  @cheenamalhotra the string for this key is `Connection parameters cannot be null`, it doesn't tell the user which field is missing, are we ok with it?

Reply:
That is certainly not sufficient.
We need the username message, which would come from here I think: 
https://github.com/microsoft/sqltoolsservice/blob/ccf95aed77898c0b0dbc804c29a486ce092cd41e/external/Microsoft.SqlTools.CoreServices/Connection/ConnectParamsExtensions.cs#L38-L45

Reply:
About the `ErrorMessage` field, I think the only other place it's assigned to is here: 

https://github.com/microsoft/sqltoolsservice/blob/ccf95aed77898c0b0dbc804c29a486ce092cd41e/external/Microsoft.SqlTools.CoreServices/Connection/ConnectionServiceCore.cs#L533-L551

Which makes this change a reasonable fix.